modules/xfem/src/userobjects/MeshCut3DUserObject.C

```diff
@@ -0,0 +1,1106 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "MeshCut3DUserObject.h"
+
+#include "XFEMFuncs.h"
+#include "MooseError.h"
+#include "libmesh/string_to_enum.h"
+#include "MooseMesh.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/edge_edge2.h"
+
+template <>
+InputParameters
+validParams<MeshCut3DUserObject>()
+{
+  InputParameters params = validParams<GeometricCutUserObject>();
+  params.addRequiredParam<std::string>("mesh_file", "Mesh file for XFEM geometric cuts");
```

Comment:
We have a mesh file type `MeshFileName`. We use it because it works with Peacock/Atom.io.